### ISSUE TITLE
`flutter build macos-framework` can't build embeddable frameworks

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#117404]
There should be an option to create a framework that works with the embedder api. 

When using `flutter build macos-framework` the resulting Flutter framework is missing the symbols required to use it with the embedder API.  Without these symbols the project cannot be loaded via the embedder api.  

## Seen results

```sh
$ flutter build macos-framework
...
$ nm build/macos/framework/Release/FlutterMacOS.xcframework/macos-arm64_x86_64/FlutterMacOS.framework/FlutterMacOS | grep FlutterEngineRun
$
```

## Expected results

```sh
$ flutter build macos-framework
...
$ nm build/macos/framework/Release/FlutterMacOS.xcframework/macos-arm64_x86_64/FlutterMacOS.framework/FlutterMacOS | grep FlutterEngineRun
000000000096c220 T _FlutterEngineRun
$
```

cc @cbracken @jmagman

**Comment by jmagman** [comment]
I don't think this is specific to `flutter build macos-framework` but to FlutterMacOS.framework in release mode.

The symbols are embedded in the Debug framework (`DEBUG_INFORMATION_FORMAT=dwarf`), but in Release it should be a separate dSYM (`DEBUG_INFORMATION_FORMAT=dwarf-with-dsym`).  But right now we don't ship the framework dSYM in the SDKs, you have to download them separately. On iOS those instructions are at https://github.com/flutter/flutter/wiki/Crashes#ios, there don't seem to be the same instructions for macOS, but it's the same, example at https://storage.cloud.google.com/flutter_infra_release/flutter/cf69289fbda1a4890d2f2d650a49403884db2992/darwin-x64-release/FlutterMacOS.dSYM.zip.  

Ideally we would ship the dSYMs in the SDK and copy them to the right spot for Xcode etc. to find.  iOS feature request at https://github.com/flutter/flutter/issues/79790, I don't see a macOS version.

**Comment by jmagman** [comment]
> Without these symbols the project cannot be loaded via the embedder api.

What does that mean, exactly?

**Comment by jmagman** [comment]
Note those symbols are present in Debug, just in Release they are (correctly) stripped in expectation that there will be a dSYM file (which doesn't exist but should).
```
$ nm build/macos/framework/Debug/FlutterMacOS.xcframework/macos-arm64_x86_64/FlutterMacOS.framework/FlutterMacOS  | grep FlutterEngineRun
0000000000972a00 t _FlutterEngineRun
00000000009743c0 t _FlutterEngineRunInitialized
000000000097b0f0 t _FlutterEngineRunTask
000000000097c4c0 t _FlutterEngineRunsAOTCompiledDartCode
```

**Comment by gaaclarke** [comment]
Sorry, it's not clear.  So there are 2 different types of the engine builds: Flutter.framework and FlutterEmbedder.framework.  The Flutter.framework will have the symbols for things like the `FlutterViewController`.  But if you are creating your own embedder those unnecessary, so there is the FlutterEmbedder.framework which has all the symbols from `embedder.h` visible.

Why have 2 different frameworks?  An optimization that makes the sizes of the frameworks smaller.

The feature request is that we should have a flag in the tool that will export a built flutter project that is intended to be used with the `embedder.h` interface.  The output should look something like this:

```
$ ls
FlutterEmbedder.xcframework
embedder.h
App.xcframework
```

Currently, the tool just assumes you are using the standard embedder and people writing their own embedders need to go off and build the engine themselves.

**Comment by jmagman** [comment]
@gaaclarke per https://github.com/flutter/flutter/wiki/Custom-Flutter-Engine-Embedders I see `flutter/<uuid>/darwin-x64/FlutterEmbedder.framework.zip`, but `darwin-x64` is the debug directory.  Where are profile and release uploaded to?

**Comment by jmagman** [comment]
This should be blocked on https://github.com/flutter/flutter/issues/111827, the `FlutterEmbedder.framework` needs to support arm to run natively on M1 Macs
